tests/api_tests/test_search.py

```diff
@@ -315,6 +317,32 @@ def test_custom_search_results(self):
         custom_score = custom_res["hits"][0]["_score"]
 
         self.assertEqual(custom_score, original_score)
+    
+    def test_opensearch_search_retry(self):
+        """
+            Ensures that retries are implemented due to transient
+            network errors occuring when sending search requests to OpenSearch.
+        """
+        self.client.create_index(self.index_name_1)
+        utilities.control_marqo_os("marqo-os", "stop")
+        
+        for search_method in ["TENSOR", "LEXICAL"]:
+            try:
+                res = self.client.index(self.index_name_1).search(
+                    q="blah",
+                    device="cpu",
+                    search_method=search_method
+                )
+            except Exception as e:
+                assert e.__class__ == MarqoWebError
+                pass
+
+            log_blob = utilities.retrieve_docker_logs("marqo")
+
+            retry_text = "BackendCommunicationError encountered... Retrying request to"
+            assert retry_text in log_blob
```

Comment:
This does not ensure retries happen the correct number of times. You can count occurrences of `retry_text` in `log_blob` to be more accurate!